hail/src/main/scala/is/hail/expr/ir/EmitFunctionBuilder.scala

```diff
@@ -445,10 +445,7 @@ class EmitFunctionBuilder[F >: Null](
   def getOrDefineMethod(prefix: String, key: Any, argsInfo: Array[TypeInfo[_]], returnInfo: TypeInfo[_])
     (f: EmitMethodBuilder => Unit): EmitMethodBuilder = {
     methodMemo.get(key) match {
-      case Some(mb) =>
-        assert(mb.parameterTypeInfo.sameElements(argsInfo))
```

Comment:
TypeInfo doesn't have an equality defined, so this assertion is bad. We should probably add an equality to TypeInfo in the future so we can assert here.